ern-api-impl-gen/src/generators/ApiImplGen.ts

```diff
@@ -79,23 +81,66 @@ export default class ApiImplGen {
     reactNativeVersion: string
   ) {
     try {
-      shell.cd(downloadPath)
+      shell.pushd(downloadPath)
```

Comment:
Better to use `pushd` along with `popd` in `finally` block rather than `cd`. With `cd` we can easily lose track of which directory we are in if not properly rolled back in `finally`. If using `cd` with `finally` we still need to back up the current directory before running `cd` (so that we can roll back in `finally`) which is cumbersome and easy to forget.